arangod/Indexes/Index.cpp

```diff
@@ -361,8 +361,36 @@ bool Index::validateId(char const* key) {
   }
 }
 
-/// @brief validate an index handle (collection name + / + index id)
-bool Index::validateHandle(char const* key, size_t* split) {
+/// @brief validate an index name
```

Comment:
Is there a specification somewhere (other than in this function) that says what an index name can be?

Reply:
They're subject to the same restrictions as collection names, so it's not a 100% direct documentation, but we do talk about it with index creation: https://github.com/arangodb/docs/blame/master/3.5/indexing-working-with-indexes.md#L103

Reply:
I was pointed towards this here: https://www.arangodb.com/docs/3.4/data-modeling-naming-conventions-collection-and-view-names.html which at least gives a specification.

Probably unrelated to this PR: Why is this file using `char *` instead of `std::string`? Is this historical?

Reply:
The reason why I am grating on at this is that I suspect that there should be validation code somewhere "central", and indeed there is some checking code in `vocbase.{h,cpp}`. It is probably not a good idea to duplicate this kind of code all over the place.

Reply:
I think you're right that we should attempt not to duplicate the actual verification code, but I do think that the method for verifying an index name should live here, with the rest of the high-level index code. It could very easily just wrap a method that lives elsewhere though. I'll look into that.

Regarding the `char*` business, I think it's a historical holdout, yes, but mostly there to allow for in-place verification regardless of where the underlying data is (string, vpack, mmfiles) without explicit wrapping in a `StringBuffer`. I think once we are able to upgrade to C++17 and have access to `std::string_view`, we can clean up most of these types of interfaces.

Reply:
@markuspf Okay, I have changed the method to wrap the code we already have.